### PR TITLE
Remove mockito and all related features

### PR DESCRIPTION
**Pull request by rubengees**
Since I also suffered from long autocompletion lists (I mainly use mockk), I decided to remove the dependency on Mockito and all related features as outlined in  https://github.com/MarkusAmshove/Kluent/issues/73.

Thus this would fix https://github.com/MarkusAmshove/Kluent/issues/73.

- [x] I've added my name to the `AUTHORS` file, if it wasn't already present.

### CODE REVIEW

**Comment by rubengees** [comment]
I have seen that `Spek` should also be removed for 2.0. Should I do this as part of this PR? Or open a new one? Also, `Spek` does not seem to be an `api` dependency, what is the rationale for removing it?

File: jvm/build.gradle [hunk 1]
```diff
@@ -21,12 +19,9 @@ dependencies {
     testImplementation libs.kotlin_test_jvm
     testImplementation libs.kotlin_test_junit
 
-
     // TODO: To be removed
```
Comment (rubengees) [inline jvm/build.gradle]: Not sure about this? Is this related only to `Mockito` and `Spek`?

**Comment by MarkusAmshove** [comment]
Thank you for the PR  :)

I still have to figure out what the right approach is for these big changes, as they will clearly break code.
I'm leaning into having a 1.x branch for the "old" stuff while 2.0 having as much Deprecated annotations with codefixes as possible and then removing the stuff in 2
1.

As for Spek, you shouldn't have to worry about it, because it *should* only be present in backtick tests.

I wanted to remove Spek to have all the tests following the same format, as Spek is jvm only.

**Comment by rubengees** [comment]
Hmm, I would propose to have a 1.x branch (and a new release 1.5 soon) with everything this PR removes marked as `@Deprecated` and keep the master branch as the baseline for 2.x (which then removes all this). I can help with the former if you like :)

Ah I see, can I help with the removal of `Spek`? That could be done right now since that is no public API in any way.

**Comment by MarkusAmshove** [comment]
Converting Spek to kotlin.test is something I would do in a separate PR, because it isn't user facing and can easily be merged now. 

If you want to go for it, don't hesitate  :-)

**Comment by MarkusAmshove** [comment]
I've now removed Spek everywhere, so we're getting close :-)

**Comment by rubengees** [comment]
Sorry, totally forgot about this! Is there something remaining to be done?

**Comment by MarkusAmshove** [comment]
Hi, as the mocking feature is still used and even improved (there's currently a PR for it), I'm currently rethinking this approach. It might be better to pull it out into its own artifact, so instead of losing the functionality, people can add another artifact and their tests aren't broken

**Comment by rubengees** [comment]
Okay, that sounds like a good option too.
I'll close this PR then.
